lib/matplotlib/dates.py

```diff
@@ -212,23 +212,24 @@ def _to_ordinalf(dt):
     days, preserving hours, minutes, seconds and microseconds.  Return value
     is a :func:`float`.
     """
-
-    if hasattr(dt, 'tzinfo') and dt.tzinfo is not None:
-        delta = dt.tzinfo.utcoffset(dt)
-        if delta is not None:
-            dt -= delta
+    # Convert to UTC
+    tzi = getattr(dt, 'tzinfo', None)
+    if tzi is not None:
+        dt = dt.astimezone(UTC)
+        tzi = UTC
 
     base = float(dt.toordinal())
-    if isinstance(dt, datetime.datetime):
-        # Get a datetime object at midnight in the same time zone as dt.
-        cdate = dt.date()
-        midnight_time = datetime.time(0, 0, 0, tzinfo=dt.tzinfo)
+
+    # If it's sufficiently datetime-like, it will have a `date()` method
+    cdate = getattr(dt, 'date', lambda: None)()
+    if cdate is not None:
+        # Get a datetime object at midnight UTC
+        midnight_time = datetime.time(0, tzinfo=tzi)
 
         rdt = datetime.datetime.combine(cdate, midnight_time)
-        td_remainder = _total_seconds(dt - rdt)
 
-        if td_remainder > 0:
-            base += td_remainder / SEC_PER_DAY
+        # Append the seconds as a fraction of a day
+        base += _total_seconds(dt - rdt) / SEC_PER_DAY
```

Comment:
We are still testing 2.6 for the 1.5.x branch but other wise yes, remove
the 2.6 related stuff.  We just will not backport this to 1.5.x (there
should not be a 1.5.2 release).

On Sat, Apr 2, 2016 at 5:31 PM Paul Ganssle <notifications@github.com>
wrote:

> In lib/matplotlib/dates.py
> <https://github.com/matplotlib/matplotlib/pull/6262#discussion_r58300209>:
>
> >
> > -        if td_remainder > 0:
> > -            base += td_remainder / SEC_PER_DAY
> > +        # Append the seconds as a fraction of a day
> > +        base += _total_seconds(dt - rdt) / SEC_PER_DAY
>
> Note: this _total_seconds function was added for Python 2.6
> compatibility, but based on the CI it seems that you've dropped Python 2.6
> support. Have there been enough Python 2.6-breaking changes that it makes
> sense to drop this compatibility artifact?
>
> —
> You are receiving this because you commented.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/6262/files/66c0ed7a019fb734de3615302923839bc692ba12#r58300209>
>